agent/config/runtime.go

```diff
@@ -48,118 +55,492 @@ type RuntimeConfig struct {
 	ConsulSerfWANSuspicionMult       int
 	ConsulServerHealthInterval       time.Duration
 
-	ACLAgentMasterToken    string
-	ACLAgentToken          string
-	ACLDatacenter          string
-	ACLDefaultPolicy       string
-	ACLDownPolicy          string
-	ACLEnforceVersion8     bool
+	// ACLAgentMasterToken is a special token that has full read and write
+	// privileges for this agent, and can be used to call agent endpoints
+	// when no servers are available.
+	//
+	// hcl: acl_agent_master_token = string
+	ACLAgentMasterToken string
+
+	// ACLAgentToken is the default token used to make requests for the agent
+	// itself, such as for registering itself with the catalog. If not
+	// configured, the 'acl_token' will be used.
+	//
+	// hcl: acl_agent_token = string
+	ACLAgentToken string
+
+	// ACLDatacenter is the central datacenter that holds authoritative
+	// ACL records. This must be the same for the entire cluster.
+	// If this is not set, ACLs are not enabled. Off by default.
+	//
+	// hcl: acl_datacenter = string
+	ACLDatacenter string
+
+	// ACLDefaultPolicy is used to control the ACL interaction when
+	// there is no defined policy. This can be "allow" which means
+	// ACLs are used to black-list, or "deny" which means ACLs are
+	// white-lists.
+	//
+	// hcl: acl_default_policy = ("allow"|"deny")
+	ACLDefaultPolicy string
+
+	// ACLDownPolicy is used to control the ACL interaction when we cannot
+	// reach the ACLDatacenter and the token is not in the cache.
+	// There are two modes:
+	//   * allow - Allow all requests
+	//   * deny - Deny all requests
+	//   * extend-cache - Ignore the cache expiration, and allow cached
+	//                    ACL's to be used to service requests. This
+	//                    is the default. If the ACL is not in the cache,
+	//                    this acts like deny.
+	//
+	// hcl: acl_down_policy = ("allow"|"deny"|"extend-cache")
+	ACLDownPolicy string
+
+	// ACLEnforceVersion8 is used to gate a set of ACL policy features that
+	// are opt-in prior to Consul 0.8 and opt-out in Consul 0.8 and later.
+	//
+	// hcl: acl_enforce_version_8 = (true|false)
+	ACLEnforceVersion8 bool
+
+	// ACLEnableKeyListPolicy ???
```

Comment:
ACLEnableKeyListPolicy is used to opt-in to the "list" policy added to KV ACLs in Consul 1.0. See https://www.consul.io/docs/guides/acl.html#list-policy-for-keys for more details.